From 54ce400764184eee067dc4f3d8d81cee2ec25537 Mon Sep 17 00:00:00 2001
From: Xaver Hugl <xaver.hugl@gmail.com>
Date: Mon, 12 Jul 2021 00:56:01 +0200
Subject: [PATCH] platforms/drm: don't delete connectors in
 DrmGpu::removeOutput

In DrmGpu::updateOutputs the connector is in the oldConnectors vector,
in DrmGpu::~DrmGpu it's in m_connectors. In both cases that's causing a
double free.

BUG: 438789
---
 src/plugins/platforms/drm/drm_gpu.cpp | 2 --
 1 file changed, 2 deletions(-)

diff --git a/src/plugins/platforms/drm/drm_gpu.cpp b/src/plugins/platforms/drm/drm_gpu.cpp
index 2e338aaf3..189c6e5d2 100644
--- a/src/plugins/platforms/drm/drm_gpu.cpp
+++ b/src/plugins/platforms/drm/drm_gpu.cpp
@@ -429,8 +429,6 @@ void DrmGpu::removeOutput(DrmOutput *output)
     emit outputRemoved(output);
     output->teardown();
     output->m_crtc = nullptr;
-    m_connectors.removeOne(output->m_conn);
-    delete output->m_conn;
     output->m_conn = nullptr;
     if (output->m_primaryPlane) {
         m_unusedPlanes << output->m_primaryPlane;
-- 
GitLab

